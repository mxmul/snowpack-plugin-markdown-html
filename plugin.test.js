/* eslint-env node, jest */
const fs = require('fs');
const path = require('path');
const glob = require('glob');

describe('snowpack-plugin-markdown-html', () => {
  describe('examples', () => {
    describe.each(glob.sync('./examples/*'))('%s', (fixture) => {
      it('emits expected index.html', () => {
        const contents = fs.readFileSync(
          path.join(fixture, './build/index.html'),
          'utf-8',
        );
        expect(contents).toMatchSnapshot();
      });
    });
  });
});
