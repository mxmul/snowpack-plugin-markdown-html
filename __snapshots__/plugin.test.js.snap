// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snowpack-plugin-markdown-html examples ./examples/basic-usage emits expected index.html 1`] = `
"<!DOCTYPE html>
    <html>
    <head>
        <title>Markdown Basic Usage</title>
    </head>
    <body>
    <h1><a id=\\"heading\\" class=\\"anchor\\" aria-hidden=\\"true\\" href=\\"#heading\\"></a>Heading</h1>
<h2><a id=\\"sub-heading\\" class=\\"anchor\\" aria-hidden=\\"true\\" href=\\"#sub-heading\\"></a>Sub-heading</h2>
<p>Paragraphs are separated
by a blank line.</p>
<p>Two spaces at the end of a line<br>
produces a line break.</p>

    </body>
    </html>"
`;

exports[`snowpack-plugin-markdown-html examples ./examples/custom-render emits expected index.html 1`] = `
"<!DOCTYPE html>
          <html>
          <head>
              <title>Markdown Basic Usage</title>
              <style>
              html, body {
                background-color: yellow;
                font-family: 'Comic Sans MS';
              }
              </style>
          </head>
          <body>
            <h1><a id=\\"heading\\" class=\\"anchor\\" aria-hidden=\\"true\\" href=\\"#heading\\"></a>Heading</h1>
<h2><a id=\\"sub-heading\\" class=\\"anchor\\" aria-hidden=\\"true\\" href=\\"#sub-heading\\"></a>Sub-heading</h2>
<p>Paragraphs are separated
by a blank line.</p>
<p>Two spaces at the end of a line<br>
produces a line break.</p>

          </body>
          </html>"
`;

exports[`snowpack-plugin-markdown-html examples ./examples/nunjucks emits expected index.html 1`] = `
"<!DOCTYPE html>
<html>
<head>
    <title>Markdown Basic Usage</title>
    <style>
    html, body {
        background-color: hotpink;
    }
    </style>
</head>
<body>
<h1><a id=\\"heading\\" class=\\"anchor\\" aria-hidden=\\"true\\" href=\\"#heading\\"></a>Heading</h1>
<h2><a id=\\"sub-heading\\" class=\\"anchor\\" aria-hidden=\\"true\\" href=\\"#sub-heading\\"></a>Sub-heading</h2>
<p>Paragraphs are separated
by a blank line.</p>
<p>Two spaces at the end of a line<br>
produces a line break.</p>

</body>
</html>
"
`;
